.github/workflows/test.yml

```diff
@@ -67,8 +67,11 @@ jobs:
           SETUPTOOLS_ENABLE_FEATURES=legacy-editable pip install -r ./requirements.txt
 
       # Must match `shard` definition in the test matrix:
-      - name: Run tests
+      - name: Run pytest tests
         run: PYTHONPATH='.' pytest --num-shards=4 --shard-id=${{ matrix.shard }} tests
+      - name: Run mypy on the test cases
+        # TODO: When mypy aligns with pyright, remove the `--no-warn-unused-ignores`.
+        run: mypy tests/python_files --no-warn-unused-ignores
```

Comment:
This is likely to create conflicts in the future, where one type checker reports an error but the error doesn't :/

Reply:
I was looking at this issue at pyright btw: https://github.com/microsoft/pyright/issues/4243